### PR TITLE
feat(registry): enforce radix/base variant parity at build time

### PR DESCRIPTION
**Pull request by okisdev**
## summary

- the registry build now fails when a component's radix and base variants drift apart on their structural contract: mismatched `data-slot` sets, mismatched exported symbols, or an npm dependency that only one tree's content actually imports
- new optional schema fields `radixDependencies` / `baseDependencies` declare style-scoped npm dependencies; they merge into `dependencies` per tree at build time, mirroring the existing `baseRegistryDependencies` semantics
- `tooltip-icon-button` moves `radix-ui` into `radixDependencies`, so `dist/base/tooltip-icon-button.json` stops installing Radix into Base UI projects (its base variant never imports it)
- no new CI wiring: the validators run inside `build-registry.ts`, which the Build Changed Packages job already executes for any PR touching `packages/ui` or `apps/registry`, and the new tests ride the existing `pnpm test` script there

### already verified

- [x] `pnpm --filter @assistant-ui/shadcn-registry test` -> 19/19 pass (covers the three validators, both dependency merge helpers, and the default/star re-export edge cases)
- [x] `pnpm --filter @assistant-ui/shadcn-registry build` -> green against the real registry; `dist/base/tooltip-icon-button.json` no longer lists `radix-ui` while `dist/tooltip-icon-button.json` still does
- [x] adversarial checks: applying the #4706 diff fails the build with the slot parity error naming the three missing slots; adding an export to `reasoning.tsx` alone fails export parity; keeping `radix-ui` in shared `dependencies` fails the dependency validator with the move to `radixDependencies` message

### reviewer should verify

- [ ] change any `data-slot` value in a paired component on one side only (for example `reasoning.tsx` without `reasoning.base.tsx`) and run `pnpm --filter @assistant-ui/shadcn-registry build`; the build should fail naming the item, file, and slot

## notes

- the drift class this guards against surfaced while reviewing #4706 (a radix-side-only ScrollArea change); that PR will fail the new slot parity check on its next CI run until its base variant is updated
- class-string-only styling drift between variants is deliberately out of scope (checking it would be all false positives); slots, exports, and dependencies are the structural contract
- `baseDependencies` has no user yet; it becomes relevant once a base variant imports `@base-ui/react` directly

<a href="https://cubic.dev/pr/assistant-ui/assistant-ui/pull/4829?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: apps/registry/scripts/build-registry.ts [hunk 1]
```diff
@@ -231,34 +234,324 @@ export function validateVariantTreesDiffer(
   }
 }
 
+function collectDataSlots(content: string) {
+  const slots = new Set<string>();
+  for (const match of content.matchAll(/data-slot="([^"]+)"/g)) {
+    slots.add(match[1]!);
+  }
```
[the remark this answers is not in the document]
Comment (okisdev) [reply apps/registry/scripts/build-registry.ts:241]: single quoted jsx attributes cannot reach the registry build in this repo: oxfmt runs with jsxSingleQuote: false and the Oxlint + Oxfmt job fails the PR before anything else, so emitted registry content is always formatter canonical (the repo has zero occurrences today). the existing BASE_VARIANT_FORBIDDEN_PATTERNS regexes rely on the same invariant, so the collector stays aligned with the file's established approach.

File: apps/registry/scripts/build-registry.ts [hunk 2]
```diff
@@ -231,34 +234,324 @@ export function validateVariantTreesDiffer(
   }
 }
 
+function collectDataSlots(content: string) {
+  const slots = new Set<string>();
+  for (const match of content.matchAll(/data-slot="([^"]+)"/g)) {
```
[the remark this answers is not in the document]
Comment (okisdev) [reply apps/registry/scripts/build-registry.ts:239]: single quoted jsx attributes cannot reach the registry build in this repo: oxfmt runs with jsxSingleQuote: false and the Oxlint + Oxfmt job fails the PR before anything else, so emitted registry content is always formatter canonical (the repo has zero occurrences today). the existing BASE_VARIANT_FORBIDDEN_PATTERNS regexes rely on the same invariant, so the collector stays aligned with the file's established approach.

File: apps/registry/scripts/build-registry.ts [hunk 3]
```diff
@@ -231,34 +234,324 @@ export function validateVariantTreesDiffer(
...
+  }
+
+  if (findings.size > 0) {
+    throw new Error(
+      `Invalid variant export parity:\n${[...findings]
+        .map((finding) => `- ${finding}`)
+        .join("\n")}`,
+    );
+  }
+}
```
[the remark this answers is not in the document]
Comment (okisdev) [reply apps/registry/scripts/build-registry.ts:442]: agreed on the duplication, deferring deliberately: the aggregate and throw skeleton is the idiom validateVariantTreesDiffer and the other #4817 validators already use, and a forEachVariantFilePair extraction should move all seven throw sites in one refactor rather than restructuring two of them here.

File: apps/registry/scripts/build-registry.ts [hunk 4]
```diff
@@ -231,34 +234,324 @@ export function validateVariantTreesDiffer(
...
+
+    if (ts.isExportAssignment(stmt)) {
+      names.add("default");
+    }
+  }
+
+  return names;
+}
+
+export function validateVariantExportParity(
```
[the remark this answers is not in the document]
Comment (okisdev) [reply apps/registry/scripts/build-registry.ts:383]: agreed on the duplication, deferring deliberately: the aggregate and throw skeleton is the idiom validateVariantTreesDiffer and the other #4817 validators already use, and a forEachVariantFilePair extraction should move all seven throw sites in one refactor rather than restructuring two of them here.
